sdk/storage/azfile/directory/models.go

```diff
@@ -69,6 +75,57 @@ func (o *DeleteOptions) format() *generated.DirectoryClientDeleteOptions {
 
 // ---------------------------------------------------------------------------------------------------------------------
 
+// RenameOptions contains the optional parameters for the Client.Rename method.
+type RenameOptions struct {
+	// FileSMBProperties contains the optional parameters regarding the SMB/NTFS properties for a file.
+	FileSMBProperties *file.SMBProperties
+	// FilePermissions contains the optional parameters for the permissions on the file.
+	FilePermissions *file.Permissions
+	// IgnoreReadOnly specifies whether the ReadOnly attribute on a pre-existing destination file should be respected.
+	// If true, rename will succeed, otherwise, a previous file at the destination with the ReadOnly attribute set will cause rename to fail.
+	IgnoreReadOnly *bool
+	// A name-value pair to associate with a file storage object.
+	Metadata map[string]*string
+	// ReplaceIfExists specifies that if the destination file already exists, whether this request will overwrite the file or not.
+	// If true, rename will succeed and will overwrite the destination file. If not provided or if false and the destination file does exist,
+	// the request will not overwrite the destination file.
+	// If provided and the destination file does not exist, rename will succeed.
+	ReplaceIfExists *bool
+	// SourceLeaseAccessConditions contains optional parameters to access the source directory.
+	SourceLeaseAccessConditions *SourceLeaseAccessConditions
```

Comment:
Should we keep `SourceLeaseAccessConditions` in directory rename options?
The service supports this header. But there no concept of lease at directory level.
x-ms-source-lease-id:<ID> | Required if the source directory has an active lease.
-- | --

https://learn.microsoft.com/en-us/rest/api/storageservices/rename-directory#request-headers
@seanmcc-msft

Reply:
Done